spec/jobs/identities_backfill_job_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+RSpec.describe IdentitiesBackfillJob, type: :job do
+  describe '#perform' do
+    # create a few rows
+    let!(:deleted) do
+      create(:service_provider_identity, :soft_deleted, :consented)
+    end
+
+    let!(:consented_at_set) do
+      create(:service_provider_identity, :consented)
+    end
+
+    let!(:no_consented_at) do
+      create(:service_provider_identity, :active)
+    end
+
+    subject { described_class.perform_now }
+
+    it 'does not update rows that have been soft-deleted' do
+      expect(deleted.deleted_at).to_not be_nil
+      expect(deleted.last_consented_at).to be_nil
+
+      subject
+      deleted.reload
+
+      expect(deleted.deleted_at).not_to be_nil
+
+      # Why is this failing?! The query should be straightforward!
+      expect(deleted.last_consented_at).to be_nil
```

Comment:
@zachmargolis or @ThatSpaceGuy Do you mind giving this a second set of eyes? This test is failing and I think I'm missing something silly.

This test is (or at least, should be) asserting that, if a row has `deleted_at` set, we do not populate `last_consented_at`. The SQL pretty simply enforces (or at least, should...) this.

I think the `deleted.reload` line is unnecessary, BTW, but I was trying things. 😇

Reply:
it's not failing on this line, its failing on line 22 (because the row is created with `:consented` so the earlier assertion is failing